org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLValidationRootSettings.java

```diff
@@ -0,0 +1,100 @@
+/**
+ *  Copyright (c) 2019 Red Hat Inc. and others.
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v2.0
+ *  which accompanies this distribution, and is available at
+ *  http://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *  Red Hat Inc. - initial API and implementation
+ */
+package org.eclipse.lemminx.extensions.contentmodel.settings;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * XMLValidationSettings
+ */
+public class XMLValidationRootSettings extends XMLValidationSettings {
+
+	private static final XMLValidationFilter[] DEFAULT_FILTERS;
+
+	private XMLValidationFilter[] filters;
+
+	static {
+		DEFAULT_FILTERS = createDefaultFilters();
+	}
+
+	public XMLValidationRootSettings() {
+		super();
+		setFilters(DEFAULT_FILTERS);
+	}
+
+	public XMLValidationFilter[] getFilters() {
+		return filters;
+	}
+
+	public void setFilters(XMLValidationFilter[] filters) {
+		this.filters = filters;
+	}
+
+	public XMLValidationSettings getValidationSettings(String uri) {
+		if (filters != null) {
+			for (XMLValidationFilter filter : filters) {
+				if (filter.matches(uri)) {
+					return filter;
+				}
+			}
+		}
+		return this;
+	}
+
+	public XMLValidationRootSettings merge(XMLValidationRootSettings settings) {
+		if (settings != null) {
+			this.filters = settings.getFilters();
+		}
+		super.merge(settings);
+		return this;
+	}
+
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = super.hashCode();
+		result = prime * result + Arrays.hashCode(filters);
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (!super.equals(obj))
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		XMLValidationRootSettings other = (XMLValidationRootSettings) obj;
+		return Arrays.equals(filters, other.filters);
+	}
+
+	private static XMLValidationFilter[] createDefaultFilters() {
+		List<XMLValidationFilter> filters = new ArrayList<>();
```

Comment:
@merks here the default validation filter that LemMinX will use by default. Do you see another filter which should be done for other files?